docs/usage/self-hosted-configuration.md

```diff
@@ -63,6 +63,36 @@ But before you disable templating completely, try the `allowedPostUpgradeCommand
 
 ## allowScripts
 
+## allowedEnv
+
+Bot administrators can allow users to configure custom environment variables within repo config.
+Only environment variables matching the list will be accepted in the [`env`](./configuration-options.md#env) configuration.
+
+Examples:
+
+```json title="renovate.json"
+{
+  "env": {
+    "SOME_ENV_VARIABLE": "some_value",
+    "EXTRA_ENV_NAME": "value"
+  }
+}
+```
+
+The above would require `allowedEnv` to be configured similar to the following:
+
+```js title="config.js"
+module.exports = {
+  allowedEnv: ['SOME_ENV_*', 'EXTRA_ENV_NAME'],
+};
+```
+
+`allowedEnv` values can be either:
+
+- An exact match string (e.g. `EXTRA_ENV_NAME`), or
+- A minimatch pattern (e.g. `SOME_ENV_*`), or
+- A regex expression (e.g. `/_ENV$/`).
+
```

Comment:
Replace with link to new docs page once https://github.com/renovatebot/renovate/pull/27387 is merged